src/borg/cache.py

```diff
@@ -379,8 +381,12 @@ def create_master_idx(chunk_idx):
             cleanup_outdated(cached_ids - archive_ids)
             if archive_ids:
                 chunk_idx = None
+                if self.progress:
+                    pi = ProgressIndicatorPercent(total=len(archive_ids), step=0.1, msg='%3.0f%% Syncing archive: %s')
```

Comment:
the percent indicator won't indicate percentage of total time needed if the uncached and cached case is by far different in runtime. it will indicate a percentage of amount of work done, though.

"%3.0f%% Syncing chunks cache. Processing archive: %s" might be slightly more precise.